src/System.CommandLine/Argument{T}.cs

```diff
@@ -160,5 +162,36 @@ public void AcceptLegalFileNamesOnly()
                 }
             });
         }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL3050", Justification = "https://github.com/dotnet/command-line-api/issues/1638")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091", Justification = "https://github.com/dotnet/command-line-api/issues/1638")]
+        internal static T? CreateDefaultValue()
+        {
+            if (default(T) is null && typeof(T) != typeof(string))
+            {
+                if (typeof(T).IsArray)
+                {
+                    return (T?)(object)Array.CreateInstance(typeof(T).GetElementType()!, 0);
+                }
+                else if (typeof(T).IsGenericType)
+                {
+                    var genericTypeDefinition = typeof(T).GetGenericTypeDefinition();
+
+                    if (genericTypeDefinition == typeof(IEnumerable<>) ||
+                        genericTypeDefinition == typeof(IList<>) ||
+                        genericTypeDefinition == typeof(ICollection<>))
+                    {
+                        return (T?)(object)Array.CreateInstance(typeof(T).GenericTypeArguments[0], 0);
```

Comment:
This isn't guaranteed to work in a NativeAOT app. Just because the app uses a `IList<double>`, it doesn't mean that `double[]`'s code is going to get preserved/generated.